platform/flowglad-next/src/subscriptions/billingRunHelpers.test.ts

```diff
@@ -1411,6 +1415,147 @@ describe('billingRunHelpers', async () => {
           billingRun.livemode
         )
       })
+      it('should grant usage credits when payment succeeds and invoice is marked as Paid', async () => {
+        // Setup: Create usage credit grant feature
+        const grantAmount = 5000
+        const feature = await setupUsageCreditGrantFeature({
+          organizationId: organization.id,
+          name: 'Test Usage Credit Grant',
+          usageMeterId: usageMeter.id,
+          renewalFrequency: FeatureUsageGrantFrequency.Once,
+          amount: grantAmount,
+          livemode: true,
+        })
+
+        await setupProductFeature({
+          organizationId: organization.id,
+          productId: product.id,
+          featureId: feature.id,
+          livemode: true,
+        })
+
+        await adminTransaction(async ({ transaction }) => {
+          const activeSubscriptionItems =
+            await selectCurrentlyActiveSubscriptionItems(
+              { subscriptionId: subscription.id },
+              billingPeriod.startDate,
+              transaction
+            )
+
+          // If no items found, update the subscription item's addedDate to be <= billing period start
+          if (activeSubscriptionItems.length === 0) {
+            const allItems = await transaction
+              .select()
+              .from(subscriptionItems)
+              .where(
+                eq(subscriptionItems.subscriptionId, subscription.id)
+              )
```

Comment:
Let's not let ORM code pollute our code, even in test files, if we can help it
```suggestion
            const allItems = await selectSubscriptionItems({ subscriptionId: subscription.id })
```